### PR TITLE
Mitaka.mitaka gateway mode.rebase.a5a0bcbc41318c25cba6632d3ebf5cd748180207

### LINKED ISSUE #878: Common networking tweaks for full feature
* Title: Common networking tweaks for full feature
* Attachments:
* Details: For enhancement

#### OpenStack Release
Mitaka+

#### Description
This is to provide both feature and tests as an enhancement for common networks feature changes.

#### Agent Version
mitaka+

#### Operating System
Any supported

#### OpenStack Release
Mitaka+

#### Deployment
Not specific to deployment outside of common-partition based networking

### PR DESCRIPTION
**Pull request by ssorenso**
@jlongstaf 
#### What issues does this address?
Fixes #878 

#### What's this change do?
A range of change of bug fixes adopted from [a suggested, outside pull request](https://github.com/F5Networks/f5-openstack-agent/pull/763l) up to [this commit](https://github.com/F5Networks/f5-openstack-agent/pull/763/commits/45e6488088a01bdb0509a9fb96838c19003f0996) functionally.

This introduces a `external_gateway_mode` feature in the `f5-openstack-agent.ini` file that will allow folks to have externally-available routes pointed at each tenant's subnets to an external route.

##### WARNING

This new feature **is not suggested** please look to our documents on the topic before considering this feature.

#### Where should the reviewer start?
This is a merge from mitaka_external_gatway_mode.  It is suggested to look at the functionality of this branch and how it differs from mitaka.

#### Any background context?
This branch has deviated from mitaka with several commits occurring in mitaka and not this branch.  @ssorenso has rebased this several times to keep this branch up to date.  Now it is a rebase back into the mitaka.

### CODE REVIEW

**Comment by ssorenso** [comment]
@richbrowne this is the deviated branch and mitaka branch rebase into mitaka.

**Comment by ssorenso** [comment]
retest this please

**Comment by ssorenso** [comment]
@pjbreaux: there's a discrepancy between smoke tests and what's in TLC for the vtep selfip.  The smoke tests have it as selfip.external versus selfip.client in tlc.

I'll fix my commit here so that it passes the test, but I thought I'd point it out.

**Comment by richbrowne** [comment]
Please rebase this PR.

**Comment by richbrowne** [comment]
Test this please

File: requirements.functest.txt [hunk 1]
```diff
@@ -6,7 +6,7 @@ git+https://github.com/openstack/neutron@mitaka-eol
 git+https://github.com/openstack/neutron-lbaas.git@mitaka-eol
 
 # F5 LBaaS dependancies
-git+https://github.com/F5Networks/f5-openstack-lbaasv2-driver.git@mitaka
```
Comment (richbrowne) [inline requirements.functest.txt]: This needs to stay at mitaka.

**Pull request merged**